lib/task_005_lib.rb

```diff
@@ -0,0 +1,110 @@
+module Task005
+  module_function
+
+  class Point
+    attr_reader :x, :y
+
+    def self.from_input(point)
+      x, y = point.split(',').map(&:to_i)
+      Point.new(x, y)
+    end
+
+    def initialize(x, y)
+      @x = x
+      @y = y
+    end
+  end
+
+  class Line
+    attr_reader :point1, :point2
+
+    def self.from_input(input)
+      point1, point2 = input.split(' -> ').map { |point| Point.from_input(point) }
+      Line.new(point1, point2)
+    end
+
+    def initialize(point1, point2)
+      @point1 = point1
+      @point2 = point2
+    end
+
+    def horizontal?
+      @point1.y == @point2.y
+    end
+
+    def vertical?
+      @point1.x == @point2.x
+    end
+
+    def horizontal_or_vertical?
+      horizontal? || vertical?
+    end
+
+    def degree_45?
+      (@point1.x - @point2.x).abs == (@point1.y - @point2.y).abs
+    end
+
+    def points_on_horizontal_or_vertical
+      x1, x2 = [@point1.x, @point2.x].sort
+      y1, y2 = [@point1.y, @point2.y].sort
```

Comment:
I would move it to separate methods like this:

```ruby
def horizontal_range
  beginning, ending = [@point1.x, @point2.x].sort
  beginning..ending
end

def vertial_range
  beginning, ending = [@point1.y, @point2.y].sort
  beginning..ending
end
```

And you could use it here and inside `points_on_degree_45` method as well ;)

Reply:
Then this method would look really simple:

```ruby
def points_on_horizontal_or_vertical
  horizontal_range.flat_map do |x|
    vertial_range.map do |y|
      [x, y]
    end
  end
end
```